library/src/extra/rocsparse_csrgeam_numeric.cpp

```diff
@@ -236,71 +239,296 @@ rocsparse_status rocsparse::csrgeam_numeric_core(rocsparse_handle             ha
+        return rocsparse_status_success;
+    }
+}
+
+template <typename T, typename I, typename J>
+rocsparse_status rocsparse::csrgeam_numeric_template(rocsparse_handle             handle,
+                                                     rocsparse_operation          trans_A,
+                                                     rocsparse_operation          trans_B,
+                                                     int64_t                      m,
+                                                     int64_t                      n,
+                                                     const void*                  alpha_device_host,
+                                                     const rocsparse_mat_descr    descr_A,
+                                                     int64_t                      nnz_A,
+                                                     const void*                  csr_val_A,
+                                                     const void*                  csr_row_ptr_A,
+                                                     const void*                  csr_col_ind_A,
+                                                     const void*                  beta_device_host,
+                                                     const rocsparse_mat_descr    descr_B,
+                                                     int64_t                      nnz_B,
+                                                     const void*                  csr_val_B,
+                                                     const void*                  csr_row_ptr_B,
+                                                     const void*                  csr_col_ind_B,
+                                                     const rocsparse_mat_descr    descr_C,
+                                                     void*                        csr_val_C,
+                                                     const void*                  csr_row_ptr_C,
+                                                     const void*                  csr_col_ind_C,
+                                                     const rocsparse_spgeam_descr descr,
+                                                     void*                        temp_buffer)
 {
-    ROCSPARSE_ROUTINE_TRACE;
+    const rocsparse_status status
+        = rocsparse::csrgeam_numeric_quickreturn(handle,
+                                                 trans_A,
+                                                 trans_B,
+                                                 m,
+                                                 n,
+                                                 (const T*)alpha_device_host,
+                                                 descr_A,
+                                                 nnz_A,
+                                                 (const T*)csr_val_A,
+                                                 (const I*)csr_row_ptr_A,
```

Comment:
Why do we cast pointers here? csrgeam_numeric_quickreturn has type-erased pointers.